src/parser.ts

```diff
@@ -181,19 +331,30 @@ export class MonsterParser {
         const damageRolls: Record<string, MeleeDamageRoll> = {};
         const rollStrings = data.damage.split(" plus ");
         for (const rollString of rollStrings) {
-            const match = rollString?.trim().match(/(\d+d\d+\+\d+) (\w*)/);
-            if (!match) return;
-            const damage = match[1];
-            const damageType = (() => {
-                const parts = match[2].split(" ").map((part) => part.toLowerCase());
-                for (const part of parts) {
-                    if (part in this.reverseDamageTypes) {
-                        return this.reverseDamageTypes[part];
+            const match = rollString?.trim().match(/(\d+?d\d+[\+\-]?\d*)+(.*)?/);
+            console.warn("roll string", rollString);
+            console.warn("match", match);
+
+            let damage;
+            let damageType;
+            if (!match) {
+                damage = rollString;
+                damageType = "untyped";
+            } else {
+                damage = match[1];
+                damageType = (() => {
+                    if (match[2]) {
+                        const parts = match[2].split(" ").map((part) => part.toLowerCase());
+                        for (const part of parts) {
+                            if (part in this.reverseDamageTypes) {
+                                return this.reverseDamageTypes[part];
+                            }
+                        }
+                    } else {
+                        return "untyped";
                     }
-                }
-
-                return "untyped";
-            })();
+                })();
+            }
```

Comment:
Instead of this nesting to handle damageType, it might be wise to roll it all back, and make it so that the immediately executed function for damageType early returns untyped if there is no match, rather than converting it to a let.

As for `damage`, you can make it a ternary if instead of this nested let stuff.

The result will be much cleaner.